kubectl-cloudflow/cmd/verify.go

```diff
@@ -0,0 +1,170 @@
+	images := config.GetNode("blueprint.images").GetObject().Items()
+
+	var imageKeyVals = make(map[string]domain.ImageReference)
+
+	for imageKey, imageRef := range images {
+		ref, err := verify.ParseImageReference(imageRef.GetString())
+		if err != nil {
+			util.LogAndExit(err.Error())
+		}
+		imageKeyVals[imageKey] = *ref
+	}
+	return imageKeyVals
+}
+
+func getFilePath(pathToBlueprint string) (string, error) {
+	blueprintURL, err := url.Parse(pathToBlueprint)
+	if blueprintURL == nil || err != nil {
+		return "", fmt.Errorf("You need to specify the full path to a blueprint file. %s, is malformed. Error: %s", pathToBlueprint, err.Error())
+	}
+	if blueprintURL.Scheme == "" { // this is a local file
+		return pathToBlueprint, nil
+	}
+
+	// url
+	timeout := time.Duration(1 * time.Second)
+	client := http.Client{
+		Timeout: timeout,
+	}
+
+	tempDir, err := getTempDir()
+
+	if err != nil {
+		return "", fmt.Errorf("Local temp dir not properly defined. Error: %s", err.Error())
+	}
+	resp, err := client.Get(pathToBlueprint)
+
+	if err != nil {
+		return "", fmt.Errorf("You need to specify the full path to a blueprint file. %s, is not accessible. Error: %s ", pathToBlueprint, err.Error())
+	}
+
+	if resp == nil || resp.StatusCode != 200 {
+		return "", fmt.Errorf("You need to specify the full path to a blueprint file. %s, is not accessible", pathToBlueprint)
+	}
+
+	file, err := ioutil.TempFile(tempDir, path.Base(blueprintURL.Path)+".")
+	if err != nil {
+		return "", fmt.Errorf("Could not create a temp file. Error: %s", err.Error())
+	}
+
+	if _, err = io.Copy(file, resp.Body); err != nil {
+		return "", fmt.Errorf("Could not copy the remote blueprint file.  Error: %s", err.Error())
+	}
+
+	if err = file.Close(); err != nil {
+		return "", fmt.Errorf("Failed to close the file.  Error: %s", err.Error())
+	}
+	return file.Name(), nil
+}
+
+func (c *verifyBlueprintCMD) verifyImpl(cmd *cobra.Command, args []string) {
+	// check blueprint contents here
+	blueprint := args[0]
+
+	file, err := getFilePath(blueprint)
+	if err != nil {
+		util.LogAndExit("Failed to fetch blueprint contents from %s", blueprint)
+	}
+
+	content, err := loadBlueprintFile(file)
+	err = verify.VerifyBlueprint(content)
+	if err != nil {
+		util.LogAndExit("Blueprint verification failed. Error: %s", err.Error())
+	} else {
+		print("Blueprint verification succeeded.")
+	}
+}
+
+func getTempDir() (string, error) {
+	tempDir, isPresent := os.LookupEnv("CLOUDFLOW_BLUEPRINT_DOWNLOAD_DIR")
+	if !isPresent {
+		return os.TempDir(), nil
+	} else {
+		if len(tempDir) == 0 {
+			return "", fmt.Errorf("You need to specify a non-empty CLOUDFLOW_BLUEPRINT_DOWNLOAD_DIR env var")
+		} else {
+			return tempDir, nil
+		}
+	}
+}
+
+func validateVerifyCmdArgs(cmd *cobra.Command, args []string) error {
+	if len(args) != 1 {
+		return fmt.Errorf("you need to specify the full path to a blueprint file")
+	}
+	blueprint := args[0]
+	blueprintURL, err := url.Parse(blueprint)
```

Comment:
This duplicates some of the code in getFilePath?

Reply:
I was looking for some way to return a value from the validation func via corba, but didnt want to have a global for this. I can remove the redundant validation code but intention here was also to fail fast at the validation phase instead of the run phase. Anyway I will fix it somehow :)